packages/grid/x-data-grid/src/hooks/utils/useGridAriaAttributes.tsx

```diff
@@ -0,0 +1,25 @@
+import { gridVisibleColumnDefinitionsSelector } from '../features/columns/gridColumnsSelector';
+import { useGridSelector } from './useGridSelector';
+import { useGridRootProps } from './useGridRootProps';
+import { gridColumnGroupsHeaderMaxDepthSelector } from '../features/columnGrouping/gridColumnGroupsSelector';
+import {
+  gridPinnedRowsCountSelector,
+  gridRowCountSelector,
+} from '../features/rows/gridRowsSelector';
+import { useGridPrivateApiContext } from './useGridPrivateApiContext';
+
+export const useGridAriaAttributes = () => {
+  const apiRef = useGridPrivateApiContext();
+  const rootProps = useGridRootProps();
+  const visibleColumns = useGridSelector(apiRef, gridVisibleColumnDefinitionsSelector);
+  const totalRowCount = useGridSelector(apiRef, gridRowCountSelector);
+  const headerGroupingMaxDepth = useGridSelector(apiRef, gridColumnGroupsHeaderMaxDepthSelector);
+  const pinnedRowsCount = useGridSelector(apiRef, gridPinnedRowsCountSelector);
+
+  return {
+    role: 'grid',
```

Comment:
Per https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/ we might have the wrong role when `treeData={true}`

```suggestion
    role: rootProps.treeData ? 'treegrid' : 'grid',
```

AG Grid seems to make it a `treegrid` in all the cases, e.g. https://ag-grid.com/react-data-grid/column-definitions/ which feels broken.

Reply:
On a related note, I would argue that `role="cell"` is wrong too, two sources:

1.

<img width="775" alt="Screenshot 2023-07-15 at 18 47 57" src="https://github.com/mui/mui-x/assets/3165635/1f42c486-2d52-438d-aeaf-1bf98c83acf9">

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/cell_role

2.

<img width="963" alt="Screenshot 2023-07-15 at 18 48 58" src="https://github.com/mui/mui-x/assets/3165635/f3ea26fe-23c2-4f92-8748-c48e38d788ff">

https://www.w3.org/WAI/ARIA/apg/patterns/grid/

Reply:
> we might have the wrong role when treeData={true}

Updated to use `treegrid` role when treeData={true} and ariaV7 experimental flag is enabled to avoid breaking changes 👍

> I would argue that role="cell" is wrong too, it should be role="gridcell"

Yup, done! I've also fixed a few accessibility tree issues and tested with VoiceOver - it's more usable now (both with `ariaV7` flag and without it) 👍